src/components/Homepage.vue

```diff
@@ -1,21 +1,52 @@
 <template>
-  <div>
-    <TweetDeck></TweetDeck>
-
-    <TwitterMsg
-      v-for="(tweet, index) in tweets"
-      :key="index"
-      :name="tweet.name"
-      :handle="tweet.handle"
-      :tsp="tweet.tsp"
-      :msg="tweet.message"
-      :id="tweet.id"
-    />
-
-    <advertisement v-on:msgFromAd="updateSomeText($event)" />
-
-    <input type="text" v-model="placeholderText" />
-  </div>
+	<div>
+		<TweetDeck></TweetDeck>
+
+		<v-menu open-on-hover offset-y>
+			<template v-slot:activator="{ on }">
+				<v-btn color="primary" dark v-on="on">
+					Filter by User
+				</v-btn>
+			</template>
+			<v-list>
+				<v-list-item
+					v-for="(name, index) in names"
+					:key="index"
+					@click="sortByAuthor(name)"
+				>
+					<v-list-item-title @click="sortByAuthor(name)">{{
+						name
+					}}</v-list-item-title>
+				</v-list-item>
+			</v-list>
+		</v-menu>
+		<div v-if="!display_sortByAuthor">
+			<TwitterMsg
+				v-for="(tweet, index) in tweets"
+				:key="index"
+				:name="tweet.name"
+				:handle="tweet.handle"
+				:tsp="tweet.tsp"
+				:msg="tweet.message"
+				:id="tweet.id"
+			/>
+		</div>
+		<div v-if="display_sortByAuthor">
+			<TwitterMsg
+				v-for="(tweet, index) in tweets_byAuthorName"
+				:key="index"
+				:name="tweet.name"
+				:handle="tweet.handle"
+				:tsp="tweet.tsp"
+				:msg="tweet.message"
+				:id="tweet.id"
+			/>
+		</div>
```

Comment:
This duplicates a lot of code.  If my eyes are right, the only difference is if we loop over `tweets` or `tweets_byAuthorName`.

Maybe make a computed property of raw tweets and determine in that code which to return.


i.e.

```
computed: {
  tweets () => {
    if (display_sortByAuthor) {
      return tweets_byAuthorName
    } else {
      return tweets
    }
}
```

Reply:
Homepage.vue is so complex I'm struggling to rewrite it w/ your recommended changes. Pls advise during a call